cosmoz-treenode-navigator.js

```diff
@@ -162,108 +163,49 @@
 		/**
 		 * Returns a node array with the children of the given path.
 		 */
-		_renderLevel: function (pl, nodes) {
-			var pathSegment = nodes,
-				pathArray,
-				level = [];
-			if (!pl) {
-				// Return the formatted root nodes.
-				level = Object.keys(nodes).map(function (key) {
-					var node = nodes[key];
-					return {
-						id: key,
-						name: node[this.comparisonProperty],
-						path: node.pathLocator,
-						children: node[this.childProperty]
-					};
-				}, this);
-				return this._sortItOut(level);
+		_renderLevel: function (pathLocator, tree) {
+			if (!tree) {
+				return;
 			}
-
-			pathArray = pl.split(this.separatorSign);
-			pathArray.forEach(function (pathKey, i, arr) {
-				var node = pathSegment[pathKey],
-					children = node[this.childProperty];
-				if (i === arr.length - 1) {
-					level = Object.keys(children).map(function (childKey) {
-						var child = children[childKey];
-						return {
-							id: child.key,
-							name: child[this.comparisonProperty],
-							path: child.pathLocator,
-							children: child[this.childProperty]
-						};
-					}, this);
-				} else {
-					pathSegment = children;
-				}
-			}, this);
-
-			return this._sortItOut(level);
+			var n = tree.getNodeByPathLocator(pathLocator),
+				level = tree.getChildren(n) ? tree.getChildren(n) : n;
+			return this._sortNodes(this._normalizeNode(level));
 		},
 		/**
-		 * Returns an Array of nodes on a given path.
+		 * Returns the "formatted" normalized node
+		 * with the properties id, name, path, sectionName, children
 		 */
-		_getNodesOnPath: function (pl, nodes) {
-			var path = pl.split(this.separatorSign),
-				pathSegment = nodes;
-				
-			return path.map(function (nodeKey) {
-				var node = pathSegment[nodeKey],
-					children = node[this.childProperty];
-				node['key'] = nodeKey;
-				if (node && children !== undefined && Object.keys(children).length > 0) {
-					pathSegment = children;
-				}
-				return node;
+		_normalizeNode: function (nodeObj) {
```

Comment:
This method actually doesn't return a formatted node as described and expected by the name but rather an array based on the object keys. It is designed to work on `node.children` input, but is not used that way.
Keep only the object remapping (1:1, return statement) in this method.
Create a new method for `_normalizeNodes()` that takes an array as input (and maps it by calling _normalizeNode() for each item). Probably needed when doing searches?
Finally, use `_normalizeNodes(tree.getChildren(node))` or even a `_normalizeChildren()` for cases where you want this code to run as it is now.